### PR TITLE
feat: search page to display random profiles

### LINKED ISSUE #6824: [FEATURE] Search page to display 5 random profiles
### Description

I think by default the search page should show 5 random profiles when there is no search performed

Tasks
- do not load all profiles on page load (think will make the page load faster)
- on page load select 5 random profiles 
- when searching perform search on the server side

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by shoebilyas123**
## Fixes Issue
Closes #6824 

## Check List (Check all the applicable boxes) 

- [ ] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots
Screenshot from 2023-05-13 20-36-23

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/6846"></a>

### CODE REVIEW

File: pages/api/users/index.js [hunk 1]
```diff
@@ -13,17 +13,25 @@ export default async function handler(req, res) {
...
-  const fileProfiles = findAllBasic();
+  let fileProfiles = findAllBasic();
+  const initialRandomUsers = req?.query?.initialRandomUsers == 'true';
+
+  const keywords = req.query?.keywords?.split(",") || [];
 
   let dbProfiles = [];
   try {
-    dbProfiles = await Profile.find({ location: { $ne: null } });
+    dbProfiles = await Profile.find({  });
```
Comment (eddiejaoude) [inline pages/api/users/index.js:23]: This will break the `maps` page

File: pages/api/users/search.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      return {
+        ...basicProfile,
+        location: profile._doc.location,
+      };
+    }
+    return basicProfile;
+  });
+
+  res.status(200).json(profiles);
+}
```
Comment (eddiejaoude) [inline pages/api/users/search.js:54]: Why is an extra api needed?

The current api can return
- list of users if no parameters
- random if query parameter set
- search if query parameter set

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you for your contribution, please see my inline comments

File: pages/api/users/index.js (hunk 1 above)
Comment (shoebilyas123) [reply pages/api/users/index.js]: Fixed.

File: pages/api/users/search.js (hunk 2 above)
Comment (shoebilyas123) [reply pages/api/users/search.js]: Sorry this is a temporary file I forgot to delete. Fixed.

File: pages/api/users/index.js [hunk 3]
```diff
@@ -23,7 +26,12 @@ export default async function handler(req, res) {
     return fileProfiles;
   }
 
-  const profiles = fileProfiles.map((fileProfile) => {
+  if(initialRandomUsers) {
```
Comment (eddiejaoude) [inline pages/api/users/index.js:29]: Please use the Prettier plugin on VScode, your coding style does not match the projects

File: pages/api/users/index.js [hunk 4]
```diff
@@ -13,7 +13,10 @@ export default async function handler(req, res) {
 
   await connectMongo();
 
-  const fileProfiles = findAllBasic();
+  let fileProfiles = findAllBasic();
+  const initialRandomUsers = req?.query?.initialRandomUsers == 'true';
```
Comment (eddiejaoude) [inline pages/api/users/index.js:17]: I don't think the `?` in `req?.` is required, also maybe use something like `const { initialRandomUsers } = req.query` and include `keywords` also

File: pages/api/users/index.js [hunk 5]
```diff
@@ -13,7 +13,10 @@ export default async function handler(req, res) {
...
+
+  const keywords = req.query?.keywords?.split(",") || [];
```
Comment (eddiejaoude) [inline pages/api/users/index.js:19]: does this need to be an array or should be treated a a whole string?

File: pages/api/users/index.js [hunk 6]
```diff
@@ -35,14 +43,37 @@ export default async function handler(req, res) {
       tags: fileProfile.tags,
     };
 
+
```
Comment (eddiejaoude) [inline pages/api/users/index.js:46]: ?

File: pages/search.js [hunk 7]
```diff
@@ -12,12 +12,12 @@ import Input from "@components/form/Input";
...
 export async function getServerSideProps() {
   let data = {
-    users: [],
+    initialRandomUsers: [],
     tags: [],
   };
   try {
-    const res = await fetch(`${process.env.NEXT_PUBLIC_BASE_URL}/api/users`);
-    data.users = await res.json();
+    const res = await fetch(`${process.env.NEXT_PUBLIC_BASE_URL}/api/users?`+new URLSearchParams({initialRandomUsers:true}));
```
Comment (eddiejaoude) [inline pages/search.js:19]: Mixed styles here...

```suggestion
    const res = await fetch(`${process.env.NEXT_PUBLIC_BASE_URL}/api/users?${new URLSearchParams({initialRandomUsers:true})}`);
```

File: pages/api/users/index.js (hunk 5 above)
Comment (shoebilyas123) [reply pages/api/users/index.js]: String for comparing usernames or names and array for comparing tags

**Pull request closed without merging**